server.js

```diff
@@ -1,9 +1,20 @@
+const taskRouter = require('./routes/tasks')
+const setMessage = require('./middleware/message')
+
+require('dotenv').config()
+const connectDB = require('./db/connect')
+const session = require('express-session')
+
 var express = require('express');
 var app = express();
 
 // set the view engine to ejs
 app.set('view engine', 'ejs');
 
+app.use(session({ secret: process.env.SESSION_SECRET, resave: false, saveUninitialized: true })); //set up the session
```

Comment:
Let's check for the presence of this secret on startup and throw an obvious error if it's not present.

```javascript
const requiredEnvVars = ['SESSION_SECRET', 'MONGODB_URI', 'SPECIAL_SUPERSECRET']
requiredEnvVars.forEach((var) => {
    if (process.env[var] === undefined) {
        throw new Error(`Required environment variable ${var} is missing`)
    }
})
```